src/components/drawer_session/JupyterLiteSessionDrawer.tsx

```diff
@@ -0,0 +1,93 @@
+import ResizableDrawer from "@exabyte-io/cove.js/dist/mui/components/custom/resizable-drawer/ResizableDrawer";
+import MessageHandler from "@exabyte-io/cove.js/dist/other/iframe-messaging";
+import JupyterLiteSession from "@exabyte-io/cove.js/dist/other/jupyterlite/JupyterLiteSession";
+import { MaterialSchema } from "@mat3ra/esse/dist/js/types";
+import { Made } from "@mat3ra/made";
+import { enqueueSnackbar } from "notistack";
+import React from "react";
+
+interface JupyterLiteTransformationProps {
+    materials: Made.Material[];
+    show: boolean;
+    onUpdate: (newMaterials: Made.Material[]) => void;
+    onHide: () => void;
+    containerRef: React.RefObject<HTMLDivElement>;
+}
+
+const DEFAULT_NOTEBOOK_PATH = "api-examples/other/materials_designer/Introduction.ipynb";
+
+class JupyterLiteSessionDrawer extends React.Component<JupyterLiteTransformationProps> {
+    messageHandler = new MessageHandler();
+
+    // eslint-disable-next-line no-useless-constructor
+    constructor(props: JupyterLiteTransformationProps) {
+        super(props);
+    }
+
+    componentDidMount() {
+        this.messageHandler.addHandlers("set-data", [this.handleSetMaterials]);
+        this.messageHandler.addHandlers("get-data", [this.returnSelectedMaterials]);
```

Comment:
Seems like code duplicate Let's reuse this through a parent

Reply:
Added base and inheritance